jadx-core/src/main/java/jadx/core/codegen/MethodGen.java

```diff
@@ -166,7 +167,13 @@ public void addInstructions(CodeWriter code) throws CodegenException {
 			code.startLine("*/");
 
 			code.startLine("throw new UnsupportedOperationException(\"Method not decompiled: ")
-					.add(mth.toString())
+					.add(mth.getParentClass().getPackage())
```

Comment:
`getPackage()` here calls `clsInfo.getAlias().getPackage()` and then returns an aliased name, so all the things seem to be nice.😃

Reply:
I mean you miss class name. Your code will print:
```java
android.support.v7.widget.measureVertical(int, int):void
```
instead:
```java
android.support.v7.widget.LinearLayoutCompat.measureVertical(int, int):void
```
And you can't just add:
```java
    .add(".")
    .add(mth.getParentClass().getAlias().getShortName())
```
because it will not work for inner classes.
So you need to implement something recursive like [here](https://github.com/skylot/jadx/blob/e0624ce986cc9a3ff55264709fed6aab6416afd9/jadx-core/src/main/java/jadx/core/dex/info/ClassInfo.java#L106)